tests/cli/test_run.py

```diff
@@ -1,210 +1,241 @@
+def test_run(generate_template, faststream_cli) -> None:
+    app_code = """
+    from faststream.asgi import AsgiFastStream, AsgiResponse, get
+    from faststream.nats import NatsBroker
 
-    with patch(
-        "faststream.cli.utils.imports._import_obj_or_factory", return_value=(None, app)
-    ):
-        result = runner.invoke(
-            faststream_app,
-            [
-                "run",
-                "faststream:app",
-                "--host",
-                "0.0.0.0",
-                "--port",
-                "8000",
-            ],
-        )
-        app.run.assert_awaited_once_with(
-            logging.INFO, {"host": "0.0.0.0", "port": "8000"}
-        )
-        assert result.exit_code == 0
+    broker = NatsBroker()
 
+    @get
+    async def liveness_ping(scope):
+        return AsgiResponse(b"hello world", status_code=200)
 
-@pytest.mark.parametrize("app", [pytest.param(AsgiFastStream())])
-def test_run_as_asgi_with_single_worker(runner: CliRunner, app: Application):
-    app.run = AsyncMock()
+    app = AsgiFastStream(broker, asgi_routes=[
+        ("/liveness", liveness_ping),
+    ])
+    """
+    with generate_template(app_code) as app_path, faststream_cli(
+        [
+            "faststream",
+            "run",
+            f"{app_path.stem}:app",
+        ],
+        extra_env={
+            "PATH": f"{app_path.parent}:{os.environ['PATH']}",
+            "PYTHONPATH": str(app_path.parent),
+        },
+    ), urllib.request.urlopen(  # nosemgrep: python.lang.security.audit.dynamic-urllib-use-detected.dynamic-urllib-use-detected
+        "http://127.0.0.1:8000/liveness"
+    ) as response:
+        assert response.read().decode() == "hello world"
+        assert response.getcode() == 200
 
-    with patch(
-        "faststream.cli.utils.imports._import_obj_or_factory", return_value=(None, app)
-    ):
-        result = runner.invoke(
-            faststream_app,
-            [
-                "run",
-                "faststream:app",
-                "--host",
-                "0.0.0.0",
-                "--port",
-                "8000",
-                "--workers",
-                "1",
-            ],
-        )
-        app.run.assert_awaited_once_with(
-            logging.INFO, {"host": "0.0.0.0", "port": "8000"}
-        )
-        assert result.exit_code == 0
 
+@pytest.mark.slow
+@pytest.mark.skipif(IS_WINDOWS, reason="does not run on windows")
+def test_run_as_asgi_with_single_worker(
+    generate_template,
+    faststream_cli,
+):
+    app_code = """
+    from faststream.asgi import AsgiFastStream, AsgiResponse, get
+    from faststream.nats import NatsBroker
+
+    broker = NatsBroker()
+
+    @get
+    async def liveness_ping(scope):
+        return AsgiResponse(b"hello world", status_code=200)
 
+    app = AsgiFastStream(broker, asgi_routes=[
+        ("/liveness", liveness_ping),
+    ])
+    """
+    with generate_template(app_code) as app_path, faststream_cli(
```

Comment:
We can't use this syntax because python3.8 and python3.8 don't support it, instead of try this:

```
with ... as ...:
    with ... as ...:
```